pkg/template/template_test.go

```diff
@@ -177,6 +177,55 @@ func TestParameterGenerators(t *testing.T) {
 	}
 }
 
+func TestProcessValue(t *testing.T) {
+	var template api.Template
+	if err := runtime.DecodeInto(kapi.Codecs.UniversalDecoder(), []byte(`{
+		"kind":"Template", "apiVersion":"v1",
+		"objects": [
+			{
+				"kind": "Service", "apiVersion": "v${VALUE}",
+				"metadata": {
+					"labels": {
+						"key1": "${VALUE}",
+						"key2": "$${VALUE}",
+						"s1_s1": "${STRING_1}_${STRING_1}",
+						"s1_s2": "${STRING_1}_${STRING_2}",
+						"i1": "${{INT_1}}"
```

Comment:
we can't return an error for that because it was valid syntax before this change, so we need to continue to tolerate it w/o throwing an error.

but i will add a test to ensure we don't mutate it.